File: Capítulo_14/Introdução à programação funcional/Programação funcional vs programação imperativa/Código em Python/functionalVsImperative.py
# Programação imperativa:
def print_array(array):
    for i in range(len(array)):
        print(array[i])

print_array([1, 2, 3, 4, 5])

# Programação functional
def for_each(array, action):
    for i in range(len(array)):
        action(array[i])

def log_item(item):
    print(item)

for_each([1, 2, 3, 4, 5], log_item)

File: Capítulo_14/Introdução à programação funcional/Programação funcional vs programação imperativa/functionalVsImperative.js
// Programação imperativa:
const printArray = function (array) {
    for (var i = 0; i < array.length; i++) {
        console.log(array[i]);
    }
};
printArray([1, 2, 3, 4, 5]);

// Programação funcional:
const forEach = function (array, action) {
    for (var i = 0; i < array.length; i++) {
        action(array[i]);
    }
};

const logItem = function (item) {
    console.log(item);
};

forEach([1, 2, 3, 4, 5], logItem);

// Sintaxe da ES2015:
const forEach_ = (array, action) => array.forEach(item => action(item));
const logItem_ = (item) => console.log(item);

forEach_([1, 2, 3, 4, 5], logItem_);